validate_sample.py

```diff
@@ -16,323 +23,106 @@
     'HH_LF3R1W', 'HH_LF3R2W', 'HH_LF3R3AW', 'HH_LF3R3BW', 'HH_LFNR1W', 'HH_LFNR2W', 'HH_LFNR3AW', 'HH_LFNR3BW',
     'HH_1LSFN', 'HH_2LEFN', 'HH_QF2R1E', 'HH_QF2R2E', 'HH_QF2R3AE', 'HH_QF3R1E', 'HH_QF3R2E', 'HH_QF3R3AE',
     'HH_QFNR1E', 'HH_QFNR2E', 'HH_QFNR3AE', 'HH_QF2R1W', 'HH_QF2R2W', 'HH_QF2R3AW', 'HH_QF3R1W', 'HH_QF3R2W',
     'HH_QF3R3AW', 'HH_QFNR1W', 'HH_QFNR2W', 'HH_QFNR3AW', 'HH_3QSFN'}
 
-
-def parse_arguments():
-    parser = argparse.ArgumentParser(description='Load a sample file into response management.')
-    parser.add_argument('sample_file_path', help='path to the sample file', type=str)
-    return parser.parse_args()
-
-
-args = parse_arguments()
-error_count = 0
-
-
-def validate_header_row(fieldnames):
-    valid_header = {'ARID', 'ESTAB_ARID', 'UPRN', 'ADDRESS_TYPE', 'ESTAB_TYPE', 'ADDRESS_LEVEL', 'ABP_CODE',
-                    'ORGANISATION_NAME', 'ADDRESS_LINE1', 'ADDRESS_LINE2', 'ADDRESS_LINE3', 'TOWN_NAME', 'POSTCODE',
-                    'LATITUDE', 'LONGITUDE', 'OA', 'LSOA', 'MSOA', 'LAD', 'REGION', 'HTC_WILLINGNESS',
-                    'HTC_DIGITAL', 'FIELDCOORDINATOR_ID', 'FIELDOFFICER_ID', 'TREATMENT_CODE',
-                    'CE_EXPECTED_CAPACITY'}
-
-    if not set(fieldnames) == valid_header:
-        print(f'Header is invalid.')
-        exit(-1)
-
-
-def validate_sample_file(sample_file_path):
+SAMPLE_ROW_SCHEMA = {
+    'ARID': [mandatory(), max_length(21), unique()],
+    'ESTAB_ARID': [mandatory(), max_length(21)],
+    'UPRN': [mandatory(), max_length(12), numeric()],
+    'ADDRESS_TYPE': [mandatory(), in_set({'HH', 'CE', 'SPG'})],
+    'ESTAB_TYPE': [mandatory(), in_set(ESTAB_TYPES)],
+    'ADDRESS_LEVEL': [mandatory(), in_set({'E', 'U'})],
+    'ABP_CODE': [mandatory(), max_length(6)],
+    'ORGANISATION_NAME': [max_length(60)],
+    'ADDRESS_LINE1': [mandatory(), max_length(60)],
+    'ADDRESS_LINE2': [max_length(60)],
+    'ADDRESS_LINE3': [max_length(60)],
+    'TOWN_NAME': [mandatory(), max_length(30)],
+    'POSTCODE': [mandatory(), max_length(8)],
+    'LATITUDE': [mandatory(), latitude_longitude(max_scale=7, max_precision=9)],
+    'LONGITUDE': [mandatory(), latitude_longitude(max_scale=7, max_precision=10)],
+    'OA': [mandatory(), max_length(9)],
+    'LSOA': [mandatory(), max_length(9)],
+    'MSOA': [mandatory(), max_length(9)],
+    'LAD': [mandatory(), max_length(9)],
+    'REGION': [mandatory(), max_length(9)],
+    'HTC_WILLINGNESS': [mandatory(), in_set({'0', '1', '2', '3', '4', '5'})],
+    'HTC_DIGITAL': [mandatory(), in_set({'0', '1', '2', '3', '4', '5'})],
+    'FIELDCOORDINATOR_ID': [max_length(7)],
+    'FIELDOFFICER_ID': [max_length(10)],
+    'TREATMENT_CODE': [mandatory(), in_set(TREATMENT_CODES)],
+    'CE_EXPECTED_CAPACITY': [numeric(), max_length(4)],
+}
+
+ValidationFailure = namedtuple('ValidationFailure', ('line_number', 'column', 'description'))
+
+
+def validate_header(header):
```

Comment:
Tiny comment but would it make sense for this to live in validators.py?

Reply:
I was keeping the sample file specific stuff in this module, if I moved this function over it would create circular imports unless I moved the `SAMPLE_ROW_SCHEMA` too, and I wanted that to live at the top of this module.